src/main/scala/temple/builder/ServerBuilder.scala

```diff
@@ -27,34 +27,34 @@ object ServerBuilder {
     detail: LanguageDetail,
     projectUsesAuth: Boolean,
   ): ServiceRoot = {
-    val language = serviceBlock.lookupMetadata[ServiceLanguage].getOrElse(ProjectConfig.defaultLanguage)
-    val database = serviceBlock.lookupMetadata[Database].getOrElse(ProjectConfig.defaultDatabase)
+    val language = serviceBlock.lookupMetadata[ServiceLanguage] getOrElse ProjectConfig.defaultLanguage
+    val database = serviceBlock.lookupMetadata[Database] getOrElse ProjectConfig.defaultDatabase
+    val readable =
+      serviceBlock.lookupLocalMetadata[Metadata.Readable] getOrElse ProjectConfig.getDefaultReadable(projectUsesAuth)
+    val writable =
+      serviceBlock.lookupLocalMetadata[Metadata.Writable] getOrElse ProjectConfig.getDefaultWritable(projectUsesAuth)
+
     val languageConfig = language match {
       case ServiceLanguage.Go => GoLanguageConfig
     }
 
     // Whether or not this service has an auth block
     val hasAuthBlock = serviceBlock.lookupLocalMetadata[ServiceAuth].isDefined
 
-    // The created by attribute exists if the project has auth and this service does not have an auth block
-    val createdBy =
-      if (projectUsesAuth && !hasAuthBlock)
-        Some(CreatedByAttribute(languageConfig.createdByInputName, languageConfig.createdByName))
-      else None
+    val createdBy = when(serviceBlock.attributes.valuesIterator contains AbstractAttribute.CreatedByAttribute) {
+      CreatedByAttribute(languageConfig.createdByInputName, languageConfig.createdByName)
+    }
```

Comment:
@smithwjv `when` is equivalent to the if-then-some-else-none. The condition has now been moved to the [validator](https://github.com/TempleEight/temple/blob/730a8c27462c98346f766e73cbfb9b4c176609a0/src/main/scala/temple/DSL/semantics/Validator.scala#L37), so it’s calculated in just one place rather than both, but otherwise the same

Reply:
Yeah I was referring to the condition, I see, fair enough